.github/workflows/benchmark-charts.yml

```diff
@@ -0,0 +1,119 @@
+name: Performance Test
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'next'
+    paths:
+      - 'packages/x-charts*/**'
+  pull_request:
+    types:
+      - labeled
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - 'master'
+      - 'next'
+
+jobs:
+  performance-test:
+    name: Benchmark Charts
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      issues: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: pnpm/action-setup@a7487c7e89a18df4991f7f222e4898a00d66ddda # v4.1.0
+        with:
+          run_install: false
+      - name: Use Node.js 22.x
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+        with:
+          node-version: 22
+          cache: 'pnpm' # https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies
+      - run: pnpm install --frozen-lockfile
+      # Ensure we are running on the prod version of our libs
+      - run: pnpm --filter "@mui/x-charts-premium..." build
+
+      - name: Run performance tests
+        run: pnpm --filter @mui-x-internal/performance-charts test:performance:ci:output
+
+      - name: Upload results
+        uses: actions/upload-artifact@v4
+        with:
+          name: charts-benchmarks-results-master.json
+          path: ./test/performance-charts/results.json
+          if-no-files-found: error
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: charts-benchmarks-results-master.json
+          path: ./results-master
+      - run: mv ./results-master/results.json ./test/performance-charts/results-master.json
+
+      - name: Run performance tests
+        uses: actions/github-script@v7
+        with:
+          script: |
```

Comment:
In `mui-public` we can the script, so we can have it typed: https://github.com/mui/mui-public/blob/master/.github/workflows/prs_check-if-pr-has-type-label.yml

eg:

```
script: |
  const script = require('./.github/workflows/scripts/prs/checkTypeLabel.js');
  await script({core, github, context});
```

https://github.com/mui/mui-public/blob/master/.github/workflows/scripts/prs/checkTypeLabel.js

Reply:
For this to work you need to checkout the repo first

Reply:
https://github.com/mui/mui-public/blob/3d1cc492da803032c8481486ef8e0e14da1a7f7e/.github/workflows/prs_check-if-pr-has-type-label.yml#L18-L24